flink-core/src/main/java/org/apache/flink/configuration/HistoryServerOptions.java

```diff
@@ -99,6 +99,11 @@
 			.withDescription("Enable HTTPs access to the HistoryServer web frontend. This is applicable only when the" +
 				" global SSL flag security.ssl.enabled is set to true.");
 
+	public static final ConfigOption<Integer> HISTORY_SERVER_MAX_SIZE =
+		key("historyserver.max.history.size")
+			.defaultValue(50)
```

Comment:
We can't enable this by default since this could result in us deleting externally kept data all of a sudden.

Reply:
So basically the default value should be `-1` if this means not clean up, right?

Reply:
Updated accordingly.